qemu/tests/create_external_snapshot_via_driver.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import re
+import shutil
+
+from autotest.client.shared import error
+
+from avocado.core import exceptions
+from avocado.utils import process
+
+from qemu.tests import qemu_disk_img_info
+
+
+def run(test, params, env):
+
+    """
+    creat external snapshot using external drivers for backing file:
+    1). Prepare enviroment of ssh/https
+    2). Boot up vm and copy a file to it, then calculate its md5 value
+    3). Create external snapshot using ssh/https driver
+    4). Boot up vm using the external snapshot
+    5). Calculate the file's md5 value again and then compare
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def clean_key():
+        process.system("rm -rf {}/id*".format(
+            params.get("ssh_key_dir")), shell=True)
+
+    @error.context_aware
+    def create_ssh_test():
+
+        """
+        Prepare ssh environment
+        :return: None
+        """
+        error.context("STEP 1: Prepare ssh login environment.")
```

Comment:
It's better to use error_context.context(). Without logging.info, it will not displayed in debug.log.